test/test_bin/test_support_stats_reporting.py

```diff
@@ -0,0 +1,95 @@
+import os
+from datetime import date
+import unittest
+from unittest.mock import patch, MagicMock
+from services.slack_service import SlackService
+from bin.support_stats_reporting import (
+    SupportRequest,
+    get_previous_working_day,
+    get_environment_variables,
+    craft_message_to_slack,
+    main,
+)
+
+
+class TestGetPreviousWorkingDay(unittest.TestCase):
+
+    def test_monday_returns_friday(self):
+        previous_day = get_previous_working_day(date_today=date(2024, 7, 8))
+        self.assertEqual(previous_day, str(date(2024, 7, 5)))
+
+    def test_tuesday_returns_monday(self):
+        previous_day = get_previous_working_day(date_today=date(2024, 7, 9))
+        self.assertEqual(previous_day, str(date(2024, 7, 8)))
+
+
+class TestGetEnvironmentVariables(unittest.TestCase):
+    @patch.dict(os.environ, {"ADMIN_SLACK_TOKEN": "test_token"})
+    def test_returns_variable(self):
+        slack_token = get_environment_variables()
+        self.assertEqual(slack_token, "test_token")
+
+    def test_raises_error_when_no_slack_token(self):
+        self.assertRaises(ValueError, get_environment_variables)
+
+
+class TestCraftMessageToSlack(unittest.TestCase):
+    def test_slack_message(self):
+
+        date_today = date(2024, 7, 16)
+        yesterdays_support_requests = [
+            SupportRequest(
+                request_type="GitHub",
+                request_action="Add user to Org",
+                request_date="2024-07-15",
+            )
+        ]
+        expected_message = "On 2024-07-15 we received 1 Support Requests: \n\n--\n*Type:* GitHub\n*Action:* Add user to Org\n*Number of requests:* 1\n"
+        self.assertEqual(
+            craft_message_to_slack(yesterdays_support_requests, date_today),
+            expected_message,
+        )
+
+    @patch.object(SlackService, "__new__")
+    def test_craft_message_to_slack(self, mock_slack_service):
+
+        mock_slack_instance = MagicMock()
+        mock_slack_service.return_value = mock_slack_instance
+
+
+class TestMain(unittest.TestCase):
+
+    def test_raises_error_when_no_slack_token(self):
```

Comment:
You don't need to test this again, this is a behaviour of the get_environment_varaibles function and not main.